### PR TITLE
refactor tryAutoGenProjectConfig [closes #1323]

### LINKED ISSUE #1323: refactor/fix `tryAutoGenProjectConfig`
## Problems

- the auto-gen never works (https://github.com/inlang/monorepo/commit/983e6eb7ae86e429ef91454ced43ec47279b0321, https://github.com/inlang/monorepo/commit/5a0ba7f5da018255f9b4d1e16bcb7d99710738b1, [discord user reports](https://discord.com/channels/897438559458430986/1084866774874468433/1150051889791119411)) 
- why? no validation in place at all if the project config works 

### `tryAutoGenProjectConfig`

Refactor the auto-gen to validate if a config works. 

- [x] remove `baseConfig`. A base config can't exist if the goal is to auto-gen the config. (Misleading communication by myself in https://github.com/orgs/inlang/discussions/1299#discussioncomment-6846447)
- [ ] remove `warnings`. @janfjohannes you mixed tryAutoGenConfig with CLI stuff here. The auto-gen function is designed for 0 user interaction. 
- [x] add validation 

**Pseudocode**

```ts
// the function either works @returns ProjectConfig or doesn't @returns undefined
export async function tryAutoGenProjectConfig(args: { fs: NodeishFilesystem }): Promise<ProjectConfig | undefined> { 

  // 1. derive potential plugins for stack of user via package.json etc.

  // 2. derive translation file locations for `pathPattern` of plugins

  // 3. stitch together a config based on 1, 2 (assume `en` as source language tag)

  // !!!4. validate the config schema with typebox

  // !!!5. validate the project with `openProject` and check the `inlang.errors` 
}
```

### cli project init command

We should reverse the order of action from trying to auto gen first, and then falling back to manual project init to increase DX and avoid confusion (I was confused why i had to select which language tags i use, expecting that project init will figure that out for me). 

```
1. tryAutoGenProjectConfig()
2. if fails: fallback to manual prompting
```

### PR DESCRIPTION
**Pull request by janfjohannes**

### CODE REVIEW

File: inlang/source-code/create-project/src/create.ts [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
  *
  * @params args.nodeishFs a nodeishFs implementation, in node this is fs:promieses
  * @params args.pathJoin implementation for joing paths, in node this should be {join} from 'path'
  * @params args.filePath optional location of your inlang config file, should be used only for testing as we hard code filepaths at lots of places
- * @params args.tryAutoGen if the module configuration should be heuristically tried to generate, if false this will be only a minimal config file
- * @params args.languageTags the languages that should be supported next to the source language
- * @params args.sourceLanguagetag the main language that other langauges are translated to
  * @returns The warnings and if successfully generated the config object, the file is also saved to the filesystem in the current path
  */
 export async function createProjectConfig(args: {
```
Comment (samuelstroschein) [inline inlang/source-code/create-project/src/create.ts:15]: I would delete `createProjectConfig` and place all logic in `tryAutoGenProjectConfig` to be explicit in what the function does: It tries to generate a project config, not to create a project.

File: inlang/source-code/create-project/src/tryAutoGenProjectConfig.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { getLanguageFolderPath } from "./getLanguageFolderPath.js"
 import type { Plugin } from "@inlang/plugin"
 import { Value } from "@sinclair/typebox/value"
 
-// FIXME: get latest major version instead
```
Comment (samuelstroschein) [inline inlang/source-code/create-project/src/tryAutoGenProjectConfig.ts:8]: [IN THE FUTURE; ignore now] @janfjohannes you can use the `@inlang/marketplace-registry` to get module links to install given that @floriandwt implements a function `getLatestVersion`

File: inlang/source-code/create-project/src/tryAutoGenProjectConfig.ts [hunk 3]
```diff
@@ -78,20 +77,26 @@ export async function tryAutoGenModuleConfig(args: {
...
+		settings: {},
+	}
 
 	const pluginId: Plugin["meta"]["id"] = `plugin.inlang.${pluginName}`
 
-	args.baseConfig.settings = {
+	config.settings = {
 		[pluginId]: { pathPattern },
-		...args.baseConfig.settings,
 	} satisfies ProjectConfig["settings"]
```
Comment (samuelstroschein) [inline inlang/source-code/create-project/src/tryAutoGenProjectConfig.ts:92]: Why arent you creating the config in one go? Is there a reason to use `config.settings = `?

**Comment by samuelstroschein** [comment]
@MaxKless is nx cloud using their own server to for compilation? If so, does the server might have RAM limitations?

**Comment by samuelstroschein** [comment]
why on earth is test running through which depends on build though? @janfjohannes feel free to merge. the website will either update or not

**Comment by brillout** [comment]
> @brillout i read a few issues that vps consumes too much memory in the build step. seems like this is the case here too

What build step exactly? You can use https://vite-plugin-ssr.com/disableAutoFullBuild to break down the build steps (which also helps for lowering peak memory usage).

**Comment by samuelstroschein** [comment]
> @brillout  What build step exactly? You can use https://vite-plugin-ssr.com/disableAutoFullBuild to break down the build steps (which also helps for lowering peak memory usage). 

`npm run build -w @inlang/website`

**Comment by samuelstroschein** [comment]
@janfjohannes let's merge this and unblock @felixhaeberle. build step will probably run through on render

**Comment by samuelstroschein** [comment]
@janfjohannes what is the status? cc @feli

**Comment by felixhaeberle** [comment]
@janfjohannes @samuelstroschein awaiting merge to publish

**Pull request merged**